mbtiles/scripts/cli.py

```diff
@@ -113,98 +132,119 @@ def mbtiles(ctx, files, output, force_overwrite, title, description,
+            else:
+                zw = int(round(math.log(360.0 / (east - west), 2.0)))
+                zh = int(round(math.log(170.1022 / (north - south), 2.0)))
+                minzoom = min(zw, zh)
+                maxzoom = max(zw, zh)
+
+            logger.debug("Zoom range: %d..%d", minzoom, maxzoom)
+
+            # Parameters for creation of tile images.
+            base_kwds.update({
+                'driver': img_format,
+                'dtype': 'uint8',
+                'nodata': 0,
+                'height': 256,
+                'width': 256,
+                'count': max(src.count, 3),
+                'crs': 'EPSG:3857'})
+
+            img_ext = 'jpg' if img_format.lower() == 'jpeg' else 'png'
+
+            # Initialize the sqlite db.
+            if os.path.exists(output):
+                os.unlink(output)
+            conn = sqlite3.connect(output)
+            cur = conn.cursor()
+            cur.execute(
+                "CREATE TABLE tiles "
+                "(zoom_level integer, tile_column integer, "
+                "tile_row integer, tile_data blob);")
+            cur.execute(
+                "CREATE TABLE metadata (name text, value text);")
 
-            # Insert tile into db.
+            # Insert mbtiles metadata into db.
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("name", title))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("type", layer_type))
             cur.execute(
-                "INSERT INTO tiles "
-                "(zoom_level, tile_column, tile_row, tile_data) "
-                "VALUES (?, ?, ?, ?);",
-                (tile.z, tile.x, tiley, buffer(contents)))
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("version", "1.1"))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("description", description))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("format", img_ext))
+            cur.execute(
+                "INSERT INTO metadata (name, value) VALUES (?, ?);",
+                ("bounds", "%f,%f,%f,%f" % (west, south, east, north)))
 
             conn.commit()
 
+            # Constrain longitude and latitude bounds.
+            EPS = 1.0e-10
+            west = max(-180 + EPS, west)
+            south = max(-85.051129, south)
+            east = min(180 - EPS, east)
+            north = min(85.051129, north)
+
+            with WarpedVRT(
+                    src, dst_crs='EPSG:3857', src_nodata=src_nodata,
+                    dst_nodata=dst_nodata, resampling=Resampling.bilinear,
+                    num_threads=num_workers) as vrt, warnings.catch_warnings():
+
+                warnings.simplefilter("ignore", category=UserWarning)
```

Comment:
There's a `NotGeoreferencedWarning` raised by rasterio at line 227 below that we never need to see. Also this silences a pair of deprecation warnings coming from rasterio that will go away by 1.0.